README.md

```diff
@@ -48,147 +48,97 @@ Hi, Yusuke!
 
 ## Emscripten integration
 
-Assume you want to run [example.c](https://github.com/mame/xterm-pty/blob/master/demo/build/example.c) in xterm.js.
+Reading user input (e.g. via functions like `fgets`) requires pausing the WebAssembly app.
 
-1. Compile it with Emscripten. Note that you need to specify `-sNO_EXIT_RUNTIME=1 -sFORCE_FILESYSTEM=1`.
+We can't block the main thread as that will prevent any events, including user input, from waking the application and causing the deadlock. Instead, we support two modes of asynchronous pausing via corresponding Emscripten features.
 
-```
-emcc -sNO_EXIT_RUNTIME=1 -sFORCE_FILESYSTEM=1 -o example-core.js example.c
-```
+### PThread proxying
 
-This will generate two files, example-core.js and example-core.wasm.
+You can compile your application with [`-s PROXY_TO_PTHREAD`](https://emscripten.org/docs/porting/pthreads.html?highlight=proxy_to_pthread#additional-flags). In this mode Emscripten will transparently move your application to run in a pthread (in a Web Worker).
```

Comment:
`-s PROXY_TO_PTHREAD` requires `-s pthread` which could be mentioned here.

Additionally, compiling with PROXY_TO_THREAD I get `ReferenceError: SharedArrayBuffer is not defined`

I'm using 9dff03f845f08a08737a5ec3da842eb2a5d659a4 and emcc 3.1.47-git (431685f05c67f0424c11473cc16798b9587bb536)

Reply:
for reference
https://github.com/astrale-sharp/7colorsC/commit/7d21af7fe74520196b09b9c4a769ea02c60b38d5

Reply:
> -s PROXY_TO_PTHREAD requires -pthread which could be mentioned here.

Fair, but then it's both documented by link and you'd get a clear hint from compiler if you forget it, so probably not a big difference.

> Additionally, compiling with PROXY_TO_THREAD I get ReferenceError: SharedArrayBuffer is not defined

This is likely because you're serving without COOP/COEP headers described below.

Reply:
Added `-pthread` to docs.